ERCS/erc-7780.md

```diff
@@ -0,0 +1,178 @@
+		 * @param hash The hash of the transaction
+		 * @param sig The signature of the transaction
+		 * @return The validation data to return to the EntryPoint
+		 */
+		function checkSignaturePolicy(
+		    bytes32 id,
+			  address sender,
+			  bytes32 hash,
+			  bytes calldata sig
+		)
+        external
+        view
+        virtual
+        returns (uint256);
+}
+```
+
+### Signer
+
+Signers MUST implement the `IModule` and the `ISigner` interface and have module type id: `6`.
+
+```solidity
+interface ISigner is IModule {
+
+		/**
+		 * @dev Check the signature of a user operation
+		 * @param id The id of the signer config
+		 * @param userOp The user operation
+		 * @param userOpHash The hash of the user operation
+		 * @return The status of the signature check to return to the EntryPoint
+		 */
+    function checkUserOpSignature(
+		    bytes32 id,
+			  PackedUserOperation calldata userOp,
+			  bytes32 userOpHash
+		)
+        external
+        payable
+        virtual
+        returns (uint256);
+
+		/**
+		 * @dev Check an ERC-1271 signature
+		 * @param id The id of the signer config
+		 * @param sender The sender of the signature
+		 * @param hash The hash to check against
+		 * @param sig The signature to validate
+		 * @return The ERC-1271 magic value if the signature is valid
+		 */
+    function checkSignature(
+		    bytes32 id,
+		    address sender,
+		    bytes32 hash,
+		    bytes calldata sig
+		)
+        external
+        view
+        virtual
+        returns (bytes4);
+}
+```
+
+### Stateless Validator
+
+Validators MUST implement the `IStatelessValidator` interface and have module type id: `7`. It is RECOMMENDED that all Validators (module type id `1`) also implement the Stateless Validator interface for additional composabillity.
+
+```solidity
+interface IStatelessValidator {
+
+		/**
+     * @dev Validates a signature given some data
+     * @param hash The data that was signed over
+     * @param signature The signature to verify
+     * @param data The data to validate the verified signature agains
+     *
+     * MUST validate that the signature is a valid signature of the hash
+     * MUST compare the validated signature against the data provided
+     * MUST return true if the signature is valid and false otherwise
+     */
+    function validateSignatureWithData(
+        bytes32 hash,
+        bytes calldata signature,
+        bytes calldata data
+    )
+        external
+        view
+        returns (bool);
+
+     /**
+     * @dev Returns boolean value if module is a certain type
+     * @param moduleTypeId the module type ID according the ERC-7579 spec
+     *
+     * MUST return true if the module is of the given type and false otherwise
+     */
+    function isModuleType(uint256 moduleTypeId) external view returns (bool);
+}
+```
+
+## Rationale
+
+TBD
+
+## Backwards Compatibility
+
+No backward compatibility issues found.
+
+## Security Considerations
+
+No security considerations found.
```

Comment:
```suggestion
TBD <!-- TODO -->
```

This is a signature validation proposal. There will absolutely be security considerations.